google/cloud/aiplatform/constants/prediction.py

```diff
@@ -0,0 +1,138 @@
+    "us-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-5:latest",
+    "europe-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-5:latest",
+    "asia-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-5:latest",
+    "us-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-5:latest",
+    "europe-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-5:latest",
+    "asia-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-5:latest",
+    "us-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-4:latest",
+    "europe-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-4:latest",
+    "asia-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-4:latest",
+    "us-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-4:latest",
+    "europe-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-4:latest",
+    "asia-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-4:latest",
+    "us-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-3:latest",
+    "europe-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-3:latest",
+    "asia-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-3:latest",
+    "us-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-3:latest",
+    "europe-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-3:latest",
+    "asia-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-3:latest",
+    "us-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-2:latest",
+    "europe-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-2:latest",
+    "asia-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-2:latest",
+    "us-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-2:latest",
+    "europe-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-2:latest",
+    "asia-docker.pkg.dev/vertex-ai/prediction/tf2-gpu.2-2:latest",
+    "us-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-1:latest",
+    "europe-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-1:latest",
+    "asia-docker.pkg.dev/vertex-ai/prediction/tf2-cpu.2-1:latest",
+    "us-docker.pkg.dev/vertex-ai/prediction/tf-cpu.1-15:latest",
+    "europe-docker.pkg.dev/vertex-ai/prediction/tf-cpu.1-15:latest",
+    "asia-docker.pkg.dev/vertex-ai/prediction/tf-cpu.1-15:latest",
+    "us-docker.pkg.dev/vertex-ai/prediction/tf-gpu.1-15:latest",
+    "europe-docker.pkg.dev/vertex-ai/prediction/tf-gpu.1-15:latest",
+    "asia-docker.pkg.dev/vertex-ai/prediction/tf-gpu.1-15:latest",
+]
+
+# Pattern:
+# {user_provided_framework: framework_name_in_uri}
+_FRAMEWORK_TO_URI_REF = {
+    SKLEARN: SKLEARN,
+    "scikitlearn": SKLEARN,
+    "scikit-learn": SKLEARN,
+    TF: TF,
+    "tensorflow": TF,
+    XGBOOST: XGBOOST,
+    "xgb": XGBOOST,
+}
+
+# Pattern:
+# {framework: (accelerator, no_accelerator)}
+_ACCELERATOR_TO_URI_REF = {
+    # If `None`, accelerator is not supported
+    SKLEARN: (None, "cpu"),
+    TF: ("gpu", "cpu"),
+    TF2: ("gpu", "cpu"),
+    XGBOOST: (None, "cpu"),
+}
+
+SERVING_CONTAINER_URIS = (
+    SKLEARN_CONTAINER_URIS + TF_CONTAINER_URIS + XGBOOST_CONTAINER_URIS
+)
+
+# All Vertex AI Prediction first-party prediction containers as a string
+_SERVING_CONTAINER_URIS_STR = " ".join(SERVING_CONTAINER_URIS)
```

Comment:
It seems like it may be simpler to build a map based on the match which will avoid having to maintain `ACCELERATOR_TO_URI_REF`, the framework constants, and this string in favor of the map.

Something like:

```Python
container_map = defaultdict(lambda: defaultdict(lambda: defaultdict(dict)))
for container_uri in SERVING_CONTAINER_URIS:
  m =  CONTAINER_URI_PATTERN.match(container_uri)
  container_map[m['region']][d['framework']][d['accelerator']][d['version']] = container_uri
```

The TF2 bookkeeping can be done here as well.

Updating with a new container uri or framework will only require adding to the list.